sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathAccessControlEntry.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+
+namespace Azure.Storage.Files.DataLake.Models
+{
+    /// <summary>
+    /// Represents an access control in a file access control list.
+    /// </summary>
+    public class PathAccessControlEntry
+    {
+        /// <summary>
+        /// Indicates whether this is the default entry for the ACL.
+        /// </summary>
+        public bool DefaultScope { get; set; }
+
+        /// <summary>
+        /// Specifies which role this entry targets.
+        /// </summary>
+        public AccessControlType AccessControlType { get; set; }
+
+        /// <summary>
+        /// Specifies the entity for which this entry applies.
+        /// Must be omitted for types mask or other.  It must also be omitted when the user or group is the owner.
+        /// </summary>
+        public string EntityId { get; set; }
+
+        /// <summary>
+        /// Specifies the permissions granted to this entry.
+        /// </summary>
+        public RolePermissions Permissions { get; set; }
+
+        /// <summary>
+        /// Empty constructor.
+        /// </summary>
+        public PathAccessControlEntry() { }
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="accessControlType">Specifies which role this entry targets.</param>
+        /// <param name="permissions">Specifies the permissions granted to this entry.</param>
+        /// <param name="defaultScope">Indicates whether this is the default entry for the ACL.</param>
+        /// <param name="entityId">Optional entity ID to which this entry applies.</param>
+        public PathAccessControlEntry(
+            AccessControlType accessControlType,
+            RolePermissions permissions,
+            bool defaultScope = false,
+            string entityId = default)
+        {
+            if (entityId != null
+                && !(accessControlType == AccessControlType.User || accessControlType == AccessControlType.Group))
+            {
+                throw new ArgumentException("AccessControlType must be User or Group if entityId is specified.");
```

Comment:
Nit - move to DataLakeErrors.cs and include the actual value (if it's easy, if not punt for now).

Reply:
I'm not sure I'm a fan of this idea.  We are only going to throw this exception in one place, and it we move it to DataLakeErrors, it will have an unwieldy name, like InvalidAccessControlTypeWithEntityIdSpecifiedException

Reply:
When I started working with your team, there was an insistence of moving every exception into *Errors.cs because it makes it easier to track down where things are being thrown from.  I don't particularly agree with that, but I do think it's weird to be inconsistent here.

Reply:
@amnguye also wants a DataLakeErrors class, I'll add it and move the exceptions.